File: frontend/src/models/auth/AuthDto.ts
import User from "@/models/auth/User";

export default class AuthDto {
  token!: string;
  user!: User;

  constructor(jsonObj?: AuthDto) {
    if (jsonObj) {
      this.token = jsonObj.token;
      this.user = new User(jsonObj.user);
    }
  }
}

File: backend/common/src/main/java/pt/ulisboa/tecnico/socialsoftware/common/dtos/auth/AuthDto.java
package pt.ulisboa.tecnico.socialsoftware.common.dtos.auth;

import java.io.Serializable;

public class AuthDto implements Serializable {
    private String token;
    private AuthUserDto user;

    public AuthDto() {
    }

    public String getToken() {
        return token;
    }

    public void setToken(String token) {
        this.token = token;
    }

    public AuthUserDto getUser() {
        return user;
    }

    public void setUser(AuthUserDto user) {
        this.user = user;
    }
}